lib/image-stream/archive.js

```diff
@@ -81,8 +80,20 @@ const extractEntryByPath = (archive, filePath, options) => {
     return Bluebird.resolve(options.default);
   }
 
-  return options.hooks.extractFile(archive, options.entries, fileEntry.name)
-    .then(rindle.extract);
+  return options.hooks.extractFile(archive, options.entries, fileEntry.name).then((stream) => {
+    return new Bluebird((resolve, reject) => {
+      let data = '';
```

Comment:
Considering that the file to be extracted might not always be a text file, what do you think about returning a buffer instead?

```js
let chunks = [];

stream.on('data', (chunk) => {
  chunks.push(chunk)
});

stream.on('end', () => {
  resolve(Buffer.concat(chunks))
});
```

Or, in case we do know the uncompressed file size beforehand, use a preallocated buffer and copy into it:

```js
let data = Buffer.alloc(uncompressedSize, 0);
let offset = 0;

stream.on('data', (chunk) => {
  chunk.copy(data, offset);
  offset += chunk.length;
});

stream.on('end', () => {
  resolve(data);
});
```